Tests/Tests.fs

```diff
@@ -162,6 +162,18 @@ type Assert with
 open FsCheck
 open FsCheck.GenOperators
 
+type ArraySegmentGenerator =
+  static member ArraySegment() =
+      {new Arbitrary<ArraySegment<int>>() with
+          override x.Generator =
+              let generator =
+                gen {
+                    let! values = Arb.generate
+                    return values |> ArraySegment<int>
+                }
+              generator
+          override x.Shrinker t = Seq.empty }

```

Comment:
this looks a bit weird, shouldn't you adjust based on t?

Reply:
Perhaps you could use Default.Array arbitrary as base?

Reply:
I tried changing to 
```
type ArraySegmentGenerator =
  static member ArraySegment() =
            Arb.Default.Array()
            |> Arb.convert ArraySegment<int> (fun (s:ArraySegment<int>) -> s.ToArray())
```
what goes wrong is that it seems that it tries to check if a value wrapped by result is equal to a value not wrapped by result.